### DISCUSSION TITLE
r/explainlikei'mfive Why cant I run InkBox on a e-ink tablet thats not officially supported?

### DISCUSSION THREAD

**Discussion by OykuInAction** [Quill-OS/quill#65] [Q&A] [answered]
Hello! I know this is a pretty dumb question, but i have virtually no experience with operating systems and what they require. My main goal for this question is to possibly get a quick rundown of what this OS needs to run on my own tablet, or any other tablet that isn't supported, as I'm willing to learn and possibly help develop it for other tablets. 

I have a boox max lumi 2, and if you aren't aware [they have broken the gpl license](http://bbs.onyx-international.com/t/install-linux-or-alternate-os-and-gpl2-kernel-source/698).  I'm curious, since you are also linux based, what gets in the way for developing your OS for Onyx in terms of what they have kept from the public about their changes? What does their law-breaking mean for you?

Also, what would I have to do to your OS to port it over myself?

Any other info about it would be great, thank you

**Comment by tux-linux** [comment, marked as answer]
Hello!

Yes, I am aware that Boox is violating the GPL license for their Linux/Android kernels, which makes it virtually impossible to do anything with them unless you're willing to get into a LOT of reverse-engineering, a thing which I don't even ever want to think about doing. To give you an idea, to replicate Boox's kernel/bootloader source code's behavior, you would probably have to figure out:
- how the RAM chip is working (although this might partly be the bootloader's job, which sources I suspect are not available either)
- how the CPU works (meaning you have to deal with it from the most low-level point possible)
- how flash storage works
- how serial output works
If you get there, you have already made a huge leap forward. But wait, that's not even all of it; you now have to figure out:
- how the screen works, how can you actually display things on it
- how the touchscreen works
- how the backlight works
- how to deal with those %$#!#@ refresh modes
Oh, and of course, if you don't want your device to literally catch on fire, you would also have to figure out how the power regulator works, especially to handle things like mere battery charging.

If we take a look at InkBox OS, it often uses a different kernel/bootloader from the vendor one with modified code in order to run properly. Even if you could theoretically get it booting if you fiddled around enough with the system's internals, you'd still have to know how the framebuffer and/or touchscreen APIs work to actually get something useful out of your device.

TL;DR: if you want to lose your mental sanity in the process, go for it

But jokes aside, thanks a lot for your interest in the project and thanks for sponsoring us. Maybe that just buying an old Kobo from the supported models list would be a better idea?
